### PR TITLE
Add new command: "clean log v2"

### PR DESCRIPTION
**Pull request by MVladislav**
# Add new command: "clean log v2"

> See more general info and details under **[discussion](https://github.com/DeebotUniverse/client.py/discussions/345)**.

- Tested with **Deebot T10** over HA - [Deebot-4-Home-Assistant](https://github.com/MVladislav/Deebot-4-Home-Assistant/tree/p95mgv) integration
- Info
  - added api for `app/dln/api/log/clean_result/list`
  - authentication is updated for use of different api prefix
  - possible this will not fully work with official api, as the request send by POST and should be a GET, i tested it with the offline bumper version

| Ecovacs Home App              | Home Assistant                                                                      |
| :---------------------------- | :---------------------------------------------------------------------------------- |
| ![clean_log_v2][clean_log_v2] | ![clean_log_v2_home][clean_log_v2_home] ![clean_log_v2_home_2][clean_log_v2_home_2] |

[clean_log_v2_home_2]: https://github.com/DeebotUniverse/client.py/assets/7400017/ef1b4c1f-8cee-4f5d-973d-e6133c4523fd
[clean_log_v2_home]: https://github.com/DeebotUniverse/client.py/assets/7400017/94cb4102-a665-4daf-986f-96df01a5949b
[clean_log_v2]: https://github.com/DeebotUniverse/client.py/assets/7400017/c9b8a5b5-0a47-40ae-a666-106df48bcded

### CODE REVIEW

File: deebot_client/commands/json/clean_logs_v2.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            "size": "10",
+            "auth": json.dumps(
+                {
+                    "with": "users",
+                    "userid": credentials.user_id,
+                    "realm": "ecouser.net",
+                    "token": credentials.token,
+                    "resource": device_info.resource,
+                }
+            ),
```
Comment (edenhaus) [inline deebot_client/commands/json/clean_logs_v2.py:47]: Are you sure that the auth must be passed as query param? Normally the authenticator will add it automatically but in the body...

Comment (MVladislav) [reply deebot_client/commands/json/clean_logs_v2.py:47]: from what the app sends as request yes, this is the communication captured over mitm:

request:

```sh
GET https://api-app.dc-eu.ww.ecouser.net/app/dln/api/log/clean_result/list?did=REPLACED_DID&res=Gy2C&country=DE&size=10&auth=REPLACED_AUTH&version=v2&logType=clean&before=1699152310000&lang=EN&et1=1699152310545&defaultLang=zh_cn&channel=google_play HTTP/2.0
country: DE
accept: application/json
sign: REPLACED_SIGN
userid: REPLACED_USER_ID
token: REPLACED_TOKEN
authorization: Bearer REPLACED_TOKEN
v: 2.4.1
appid: ecovacs
plat: android
lang: EN
content-type: application/json
ts: 1699152310572
ucid: REPLACED_UC_ID
accept-encoding: gzip
user-agent: okhttp/4.9.1
content-length: 0
```

response:

```sh
HTTP/2.0 200
content-type: application/json; charset=utf-8
date: Sun, 05 Nov 2023 02:45:10 GMT
content-length: 40

{"code":0,"message":"success","data":[]}
```

request as curl:

```sh
curl -H 'country: DE' -H 'accept: application/json' -H 'sign: REPLACED_SIGN' -H 'userid: REPLACED_USER_ID' -H 'token: REPLACED_TOKEN' -H 'authorization: Bearer REPLACED_TOKEN' -H 'v: 2.4.1' -H 'appid: ecovacs' -H 'plat: android' -H 'lang: EN' -H 'content-type: application/json' -H 'ts: 1699152310572' -H 'ucid: REPLACED_UC:ID' --compressed -H 'user-agent: okhttp/4.9.1' 'https://api-app.dc-eu.ww.ecouser.net/app/dln/api/log/clean_result/list?did=REPLACED_DID&res=Gy2C&country=DE&size=10&auth=REPLACED_AUTH&version=v2&logType=clean&before=1699152310000&lang=EN&et1=1699152310545&defaultLang=zh_cn&channel=google_play'
```

Comment (edenhaus) [reply deebot_client/commands/json/clean_logs_v2.py:47]: What happen if we omit the auth in the query params?
Does the request still work?

**Comment by MVladislav** [comment]
closed because of not needed, see also https://github.com/DeebotUniverse/Deebot-4-Home-Assistant/discussions/464

**Pull request closed without merging**